core/src/main/java/org/cru/contentscoring/core/servlets/CopyScoresToTagsServlet.java

```diff
@@ -160,12 +162,29 @@ private void copyScoreTagToPrimaryExperienceFragment(
             String primaryExperienceFragmentPath = pageContent.getValueMap().get(PRIMARY_XF_NAME, String.class);
 
             if (!Strings.isNullOrEmpty(primaryExperienceFragmentPath)) {
-                Resource experienceFragment = resourceResolver.getResource(primaryExperienceFragmentPath);
+                Resource primaryExperienceFragment = resourceResolver.getResource(primaryExperienceFragmentPath);
+                Resource experienceFragment;
+
+                if (primaryExperienceFragment != null) {
+                    if (isExperienceFragmentVariation(primaryExperienceFragment)) {
+                        experienceFragment = primaryExperienceFragment.getParent();
+                    } else if (isExperienceFragment(primaryExperienceFragment)) {
+                        experienceFragment = primaryExperienceFragment;
+                    } else {
+                        return;
+                    }
 
-                if (experienceFragment != null) {
                     Resource experienceFragmentContent = getJcrContent(experienceFragment);
                     Set<Tag> tags = buildTagsWithScore(experienceFragmentContent, tagManager, score);
                     tagManager.setTags(experienceFragmentContent, tags.toArray(new Tag[0]));
+
+                    for (Resource child : experienceFragment.getChildren()) {
```

Comment:
are all children guaranteed to be variants?

Reply:
Not sure if it's 100% guaranteed. I haven't seen anything else that is a Page as a child of an experience fragment that also has a `jcr:content` child of its own besides a variant. I could add a check if you'd like.

Reply:
I dunno. Your call, I guess.